python/src/iceberg/utils/schema_conversion.py

```diff
@@ -0,0 +1,451 @@
+        elif logical_type == "map":
+            return self._convert_logical_map_type(avro_logical_type)
+        elif logical_type in LOGICAL_FIELD_TYPE_MAPPING:
+            return LOGICAL_FIELD_TYPE_MAPPING[logical_type]
+        else:
+            raise ValueError(f"Unknown logical type: {avro_logical_type}")
+
+    def _convert_logical_decimal_type(self, avro_type: Dict[str, Any]) -> DecimalType:
+        """
+        Args:
+            avro_type: The Avro type
+
+        Examples:
+            >>> from iceberg.utils.schema_conversion import AvroSchemaConversion
+            >>> avro_decimal_type = {
+            ...     "type": "bytes",
+            ...     "logicalType": "decimal",
+            ...     "precision": 19,
+            ...     "scale": 25
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_logical_decimal_type(avro_decimal_type)
+            >>> expected = DecimalType(
+            ...     precision=19,
+            ...     scale=25
+            ... )
+            >>> actual == expected
+            True
+
+        Returns:
+            A Iceberg DecimalType
+        """
+        return DecimalType(precision=avro_type["precision"], scale=avro_type["scale"])
+
+    def _convert_logical_map_type(self, avro_type: Dict[str, Any]) -> MapType:
+        """
+        In the case where a map hasn't a key as a type you can use a logical map to
+        still encode this in Avro
+
+        Args:
+            avro_type: The Avro Type
+
+        Examples:
+            >>> from iceberg.utils.schema_conversion import AvroSchemaConversion
+            >>> avro_type = {
+            ...     "type": "array",
+            ...     "logicalType": "map",
+            ...     "items": {
+            ...         "type": "record",
+            ...         "name": "k101_v102",
+            ...         "fields": [
+            ...             {"name": "key", "type": "int", "field-id": 101},
+            ...             {"name": "value", "type": "string", "field-id": 102},
+            ...         ],
+            ...     },
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_logical_map_type(avro_type)
+            >>> expected = MapType(
+            ...         key_id=101,
+            ...         key_type=IntegerType(),
+            ...         value_id=102,
+            ...         value_type=StringType(),
+            ...         value_is_optional=False
+            ... )
+            >>> actual == expected
+            True
+
+        .. _Apache Iceberg specification:
+            https://iceberg.apache.org/spec/#appendix-a-format-specific-requirements
+
+        Returns:
+            The logical map
+        """
+        fields = avro_type["items"]["fields"]
+        assert len(fields) == 2, f"Expected two fields in the logical map, but got: {fields}"
+        key = self._convert_field(fields[0])
+        value = self._convert_field(fields[1])
```

Comment:
This is probably okay, but I don't think that there's a guarantee that the fields are in order. It would be good to check the field names.

Reply:
Replaced it by:
```python
key = self._convert_field(list(filter(lambda f: f["name"] == "key", fields))[0])
value = self._convert_field(list(filter(lambda f: f["name"] == "value", fields))[0])
```